include/deal.II/base/work_stream.h

```diff
@@ -747,29 +747,146 @@ namespace WorkStream
 
         tf::Task last_copier;
 
+        struct CopyChunk
+        {
+          std::vector<std::unique_ptr<CopyData>> copy_datas;
+
+          CopyChunk(const unsigned int chunk_size)
+          {
+            copy_datas.reserve(chunk_size);
+          }
+        };
+
         // idx is used to connect each worker to its copier as communication
         // between tasks is not supported. It does this by providing a unique
         // index in the vector of pointers copy_datas at which the copy data
         // object where the work done by work task #idx is stored
         unsigned int idx = 0;
 
-        std::vector<std::unique_ptr<CopyData>> copy_datas;
+        std::vector<CopyChunk> copy_chunks;
```

Comment:
shouldn't this be a ``std::vector<std::unique_ptr<Chunk>>`` where each ``Chunk`` contains an ``std::vector<CopyData>``?